tools/AzureRM.Example.psm1

```diff
@@ -9,6 +9,39 @@
 $PSDefaultParameterValues.Clear()
 Set-StrictMode -Version Latest
 
+if (%ISAZMODULE%)
+{
+    if ($PSVersionTable.PSVersion -lt [Version]'5.1')
+    {
+        throw "PowerShell versions lower than 5.1 are not supported in Az. Please upgrade to PowerShell 5.1 or higher."
+    }
+
+    if (Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full\' | Get-ItemPropertyValue -Name Release | Foreach-Object { $_ -lt 461808 })
```

Comment:
I would put this version detection predicate into a function, and ensure that any errors in reading the registry results in no exception.

Also, have we tested this cross platform?